cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramStopSubscriberService.java

```diff
@@ -155,6 +159,9 @@ private void processNotification(Notification notification, StructuredTableConte
     // in both situations, there isn't any program to actual stop, and the right thing to do is
     // transition the program to the killed state
     if (runtimeInfo == null) {
+      LOG.info("Could not find a program controller for application {} program {} run {}. " +
+                 "Writing a message that the program state should be killed.",
+               programRunId.getApplication(), programRunId.getProgram(), programRunId.getRun());
```

Comment:
why is this specifically `LOG.info` and the above is `LOG.debug`?

Reply:
I put the messages that result in a state change as info message. The one above is no-op, so have it as debug.